### PR TITLE
feat(perf-tests): Add performance tests using common driver perf queries

### PR DESCRIPTION
**Pull request by adambollen**
### Description

BT-5066

This PR adds performance tests that are very similar to the ones already added for `fauna-dotnet`; the tests consume a set of queries defined in the `fauna/driver-perf-utils` repo, runs them in a Concourse pipeline every 6h and outputs the statistics of the driver "overhead" to a private Slack channel.

These tests currently lack some parity with the .NET tests due to not having access to a stats collector on the client; once that's added, I can go back and improve the pagination tests.

I'll also be honest that I'm not a Java coder, so I warrant there may be some assumptions/approaches that could be improved - feel free to provide that feedback, but unless the errors are egregious, we should probably open a ticket and track fixing later.

### Motivation and context

See above

### How was the change tested?

Plenty of local testing with `gradlew test`; also reusing a lot of learnings from the .NET version of the tests.

### Change types

* - [ ] Bug fix (non-breaking change that fixes an issue)
* - [x] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [ ] My code follows the code style of this project.
* - [ ] My change requires a change to Fauna documentation.
* - [ ] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by findgriffin — DISMISSED** [review]
Looks great! Just some minor comments.

File: concourse/tasks/perf.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+image_resource:
+  type: registry-image
+  source:
+    repository: eclipse-temurin
+    tag: 17-noble
+
+params:
+  FAUNA_ENDPOINT:
+  FAUNA_SECRET:
+  FAUNA_ENVIRONMENT:
```
[the remark this answers is not in the document]
Comment (adambollen) [reply concourse/tasks/perf.yml:12]: YAML-fu.  :)  This syntax means those params are expected from the pipeline YAML and don't have defaults (which is fine).

File: src/test/java/com/fauna/perf/MetricsHandler.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        List<String> headers = List.of("ts,metric,roundTrip,queryTime,diff,tags");
+        Files.write(Paths.get(rawStatsFilename), headers, StandardOpenOption.CREATE);
+
+        List<String> blockHeaders = List.of(
+            String.format("%-35s%9s%9s%9s%9s", "TEST", "P50", "P95", "P99", "STDDEV"),
+            new String(new char[71]).replace("\0", "-")
+        );
+        Files.write(Paths.get(statsBlockFilename), blockHeaders, StandardOpenOption.CREATE);
+
+        for (Map.Entry<String, List<TestTimings>> entry : new TreeMap<>(metricsCollector).entrySet()) {
```
[the remark this answers is not in the document]
Comment (adambollen) [reply src/test/java/com/fauna/perf/MetricsHandler.java:39]: Yep, the order without was making me grumpy.  I actually need to do the same in .NET, but the order there is at least a little more consistent.

File: src/test/java/com/fauna/perf/PerformanceTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            }
+        }
+    }
+
+    private Query getCompositedQueryFromParts(List<String> parts) {
+        if (parts.size() == 1) {
+            return fql(parts.get(0));
+        }
+
+        return fql(String.join("", parts));
```
[the remark this answers is not in the document]
Comment (adambollen) [reply src/test/java/com/fauna/perf/PerformanceTest.java:105]: Fair; I'll probably just tweak it to skip if length == 0.

File: src/test/java/com/fauna/perf/testdata/TestDataParser.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.provider.Arguments;
+
+public class TestDataParser {
+    public static Stream<Arguments> getQueriesFromFile() throws IOException {
+        String content = new String(Files.readAllBytes(Paths.get("./perf-test-setup/queries.json")));
+        ObjectMapper mapper = new ObjectMapper();
+        TestQueries testQueries = mapper.readValue(content, TestQueries.class);
```
[the remark this answers is not in the document]
Comment (adambollen) [reply src/test/java/com/fauna/perf/testdata/TestDataParser.java:15]: Cool, I'll give that a look.  Thanks!

**Review by findgriffin — APPROVED** [review]

**Pull request merged**